doc/dev/tests.md

```diff
@@ -266,6 +267,14 @@ From your terminal run the `pytest` command to run all the tests that you have w
 
 Your update should run smooth and have green dots representing passing tests. Now if you look at the contents of your `tests` directory there should be a new directory called `recording` with four `.yaml` files. Each `yaml` file is a recording for a single test. To run a test in playback mode change the `testsettings_local.cfg` to `live-mode: false` and rerun the tests with the same command. The test infrastructure will use the automatically created `.yaml` recordings to mock the HTTP traffic and run the tests.
 
+### Purging Secrets
+
+The `yaml` files created from running tests in live mode store the request and response interactions between the library and the service and this can include authorization, account names, shared access signatures, and other secrets. The recordings are included in our public GitHub repository, making it important for us to remove any secrets from these recordings before committing them to the repository. There are two easy ways to remove secrets. The first is the `PowerShellPreparer` implementation, discussed above. This method will automatically purge the keys with the provided fake values. The second way is to use the `self.scrubber.register_name_pair(key, fake_key)` method, which is used when a secret is dynamically created during a test. For example, [KeyVault](https://github.com/Azure/azure-sdk-for-python/blob/9091e0ea0ddadc349cdd0f92579e155c7f44220b/sdk/keyvault/azure-keyvault-administration/tests/test_access_control.py#L43-L56) uses this method to remove GUIDs and service principals from recordings.
```

Comment:
Might be worth noting that `self.scrubber.register_name_pair(key, fake_key)` comes from AzureTestCase